libvast/vast/format/json/field_selector.hpp

```diff
@@ -36,44 +44,49 @@ struct field_selector {
       // Keep a list of failed keys to avoid spamming the user with warnings.
       if (unknown_types.insert(field).second)
         VAST_WARN("{} does not have a layout for {} {}",
-                  detail::pretty_type_name(this), Specification::field, field);
+                  detail::pretty_type_name(this), field_name_, field);
       return {};
     }
     return it->second;
   }
 
-  caf::error schema(const vast::schema& s) {
+  inline caf::error schema(const vast::schema& s) override {
     for (const auto& t : s) {
-      auto sn = detail::split(t.name(), ".");
```

Comment:
Note for the reviewer: I fixed a bug where this was broken for event types containing more than one `.` in the name, e.g., where the type prefix is `foo.bar`, the field at the given field name contains `bad` and the event type is `foo.bar.baz`. This was broken before this change.